mlrun/artifacts/base.py

```diff
@@ -88,9 +88,10 @@ class ArtifactSpec(ModelObj):
         "db_key",
         "extra_data",
         "unpackaging_instructions",
+        "producer",
```

Comment:
Why was this change needed? This means the `producer` will now be in the result of `base_dict()` for artifacts - is this intended?

Reply:
When setting an artifact on the project it is using the `base_dict()`, so the `producer` property was not exported because it was a part of the `_extra_fields`.
This is a crucial part to maintain the producers when loading a project - to actually have it on the exported spec.

Reply:
So this means that this fix will not work for any project exported prior to this PR, right?
Also, need to make sure there are no BC issues there (I don't think there are, but needs to be verified).

Reply:
It obviously won't fix projects that were exported before this fix, so I'm not sure what BC issues we should check here..

Reply:
It's not obvious that projects exported before the fix won't be fixed - it's the situation because the export doesn't contain the information needed for the fix.
As for BC, just to make sure that now that artifact export/import contains the producer field in the base export fields, that artifacts that were exported before won't fail importing due to it (again, I'm pretty sure they won't fail, just something to pay attention to).